src/core/event/poll_group.cpp

```diff
@@ -61,10 +69,39 @@ poll_group::poll_group(const struct xlio_poll_group_attr *attr)
     m_event_handler = std::make_unique<event_handler_manager_local>();
     m_tcp_timers = std::make_unique<tcp_timers_collection>(1U);
     m_tcp_timers->set_group(this);
+
+    s_poll_groups_lock.lock();
+    s_poll_groups.push_back(this);
+    s_poll_groups_lock.unlock();
+
+    grp_logdbg("Polling group %p created", this);
 }
 
 poll_group::~poll_group()
 {
+    s_poll_groups_lock.lock();
+    auto iter = std::find(s_poll_groups.begin(), s_poll_groups.end(), this);
+    if (iter != std::end(s_poll_groups)) {
+        s_poll_groups.erase(iter);
+    }
+    s_poll_groups_lock.unlock();
+
+    grp_logdbg("Polling group %p destroyed", this);
+}
+
+/*static*/
+void poll_group::destroy_all_groups()
+{
+    poll_group *grp;
+
+    do {
```

Comment:
Can just copy the list to another list with one line and then iterate the new list with for (xx:xx). It doesn't have to be efficient anyway and will very simplify the code.

Reply:
done.